QEApp Documentation (#414)
aiidalab/aiidalab-qe: docs/source/tutorials/advance_run_on_remote_computer.rst

.. _run_calculation_on_remote_computer:

Run calculation on remote computer
==================================

> This time, the goal is to submit a band structure calculation of magnetic cobalt on a remote machine (i.e., on a different computational resource). The instruction in collapsible text is for CSCS machine.

Reviewer: There is actually no collapsible text,  and the text is stil very CSCS specific, can this be fixed?

Step 1 Select a structure
-------------------------

Again use the "From examples" structure importer and select cobalt.

Step 2 Configure workflow
-------------------------

Turn on the `Magnetism` option.

> For protocol, please select "fast" (or select the "precise" protocol if you have enough resource to test).

Reviewer: ```suggestion
For protocol, please select "fast" (or select the "precise" protocol if you have enough resource for this test).
```

Step 3 Choose computational resources
-------------------------------------

Now you need set up a new computer (e.g., the CSCS Daint or Eiger machine if you have acount on it) and the codes on this newly setup computer (i.e., you need to tell AiiDA where the executables are, which modules to load etc.).

.. note::

   The widget to assist you in the computer/code setup is not very convinient for CSCS machines, because CSCS requires to specify the account and setting up a default one is not currently straightforward in AiiDA; we are working to improve this.

**Setup new code**
- Click the button `Setup new code`.
- In the `Domain` field, please select `daint.cscs.ch`.
- Click `Detailed Setup`, then it will show three steps:

>     - password-less SSH
>     - computer in AiiDA
>     - code in AiiDA

Reviewer: Let's be more explicit, e.g. I provide here an example:
```suggestion
    - setup a password-less SSH connection to the computer (this needs to be done only once)
    - setup the remote computer/cluster in AiiDA
    - setup the codes you need to use (e.g. the Quantum ESPRESSO executables) in AiiDA
```

> **Set up password-less SSH connection**

Reviewer: Shall we use headers instead of just bold?

- In the `SSH username`, input your username for CSCS.
- Click `Setup ssh`.
- On the textbox above `Detailed Setup`, input your password, and click `Continue`.

> - Wait a few seconds, it will ask you to input your password again, and click `Continue`

Reviewer: Shall we add some screenshots in this section?

If you see a message as below:
```
30s timeout ...
```
Please click the `Setup ssh` button again and input the password.

**Setup a computer in AiiDA**

> Click the button `Setup computer`, then add one line in the `Prepend text` to specify your account (it is the project account, such as `mr32` or `mrcloud`) at CSCS. Replace the `<account_name>` string with it!

Reviewer: Make this sentence less CSCS-centric. E.g. you can write "I you use SLURM as the scheduler and you need to configure explicitly an account to run the simulations, please specify the following line in the `Prepend text`, replacing `<account_name>` with the name of the account or project that you want to use for your simulations.

```
#SBATCH --account=<account_name>
```
You can find further parameters in the screenshot below.

.. figure:: /_static/images/step1_select_structure.png
   :width: 12cm

Click the button `test computer`, then wait 1 minute: you will see this log message if everything was setup correctly:

.. code-block:: none

   Report: Testing computer<daint-mc> for user<aiida@localhost>...
   * Opening connection... [OK]
   * Checking for spurious output... [OK]
   * Getting number of jobs from scheduler... [OK]: 2255 jobs found in the queue
   * Determining remote user name... [OK]: xingwang
   * Creating and deleting temporary file... [OK]
   Success: all 5 tests succeeded

**Set up a code**

> In the field `Select computer`, please select the computer you just setup, in this case `daint-mc`. Check all other options, and then click the button `Setup code` to setup the `pw.x` code.

Reviewer: ```suggestion
In the field `Select computer`, please select the computer you just setup. Check all other options, and then click the button `Setup code` to setup the Quantum ESPRESSO `pw.x` code. Note that you need to have the `pw.x` code already compiled on the supercomputer. You can refer t the screenshot below for an example.
```

.. figure:: /_static/images/setup_computer.png
   :width: 12cm

Similarly, setup the `dos.x` and `projwfc.x` codes, needed by the app to compute DOS and PDOS. You can skip the computer setup step, since it needs to be done only once per comouter, and go to `Set up a code in AiiDA` directly to setup the remaining codes.

> After you finishing the codes setup, you can launch a new calculation with the newly setup codes, that should be called:
> 
> - pw-7.0@daint-mc
> - dos-7.0@daint-mc
> - projwfc-7.0@daint-mc

Reviewer: ```suggestion
After you finishing the codes setup, you can launch a new calculation with the newly setup codes, that should be called as ``codelabel@computerlabel``; e.g. for a computer called ``daint-mc``, the codes might be called:

- pw-7.0@daint-mc
- dos-7.0@daint-mc
- projwfc-7.0@daint-mc
```

.. figure:: /_static/images/setup_code.png
   :width: 12cm

> In the end, click the `Submit` button.

Reviewer: ```suggestion
In the end, in the main Quantum ESPRESSO app after selecting the codes, click the `Submit` button.
```

> Step 4 Check the status and results.
> ------------------------------------

Reviewer: ```suggestion
Step 4 Check the status and results
-----------------------------------
```

> 

Reviewer: Add a sentence to say "follow the same instructions as in the previous tutorial" + link
